### DISCUSSION TITLE
hardhatToken.waitForDeployment is not a function

### DISCUSSION THREAD

**Discussion by 6be709c0** [NomicFoundation/hardhat#4003] [General]
Following the Testing contracts: https://hardhat.org/tutorial/testing-contracts#5.-testing-contracts

I'm getting an error because waitForDeployment is not a function.

Within the deployTokenFixture:
```
 async function deployTokenFixture() {
    // Get the Signers here.
    const [owner, addr1, addr2] = await ethers.getSigners();

    // To deploy our contract, we just have to call ethers.deployContract and await
    // its waitForDeployment() method, which happens once its transaction has been
    // mined.
    const hardhatToken = await ethers.deployContract("Token");

    console.log(hardhatToken);
    await hardhatToken.waitForDeployment();

    // Fixtures can return anything you consider useful for your tests
    return { hardhatToken, owner, addr1, addr2 };
  }
```

Here are the available keys within hardhatToken object:
```
[
  'interface',
  'provider',
  'signer',
  'callStatic',
  'estimateGas',
  'functions',
  'populateTransaction',
  'filters',
  '_runningEvents',
  '_wrappedEmits',
  'address',
  'resolvedAddress',
  'balanceOf(address)',
  'name()',
  'owner()',
  'symbol()',
  'totalSupply()',
  'transfer(address,uint256)',
  'balanceOf',
  'name',
  'owner',
  'symbol',
  'totalSupply',
  'transfer',
  'deployTransaction'
]
```

Looks like something has been updated and waitForDeployment isn't needed anymore? The tests are all passing without it.

**Comment by fvictorio** [comment]
I think the answer here is the same as in https://github.com/NomicFoundation/hardhat/discussions/4004#discussioncomment-6119717

**Comment by flymauntains** [comment]
Hi, I am going to deploying contract to ethereum sepolia testnet using hardhat.
So, after `await contract.waitForDeployment()`, there is no change. Help me.
![Screenshot 2024-07-03 231148](https://github.com/NomicFoundation/hardhat/assets/172695995/7a17ba09-3800-4cb3-95f3-632e8be68743)
